### PR TITLE
large file upload 

### PR DESCRIPTION
**Pull request by Abhinav-grd**

### CODE REVIEW

File: src/services/uploadService.ts [hunk 1]
```diff
@@ -34,6 +46,12 @@ interface UploadURL {
     objectKey: string;
 }
 
+interface MultipartUploadURLs {
+    objectKey: string;
+    partUrls: string[];
```
Comment (vishnukvmd) [inline src/services/uploadService.ts:51]: `partURLs`

File: src/services/uploadService.ts [hunk 2]
```diff
@@ -131,6 +149,7 @@ class UploadService {
             try {
                 await this.fetchUploadURLs(token);
             } catch (e) {
+                console.log('error fetching uploadURLs', e);
```
Comment (vishnukvmd) [inline src/services/uploadService.ts:152]: `console.error(...)`.

File: src/services/uploadService.ts [hunk 3]
```diff
@@ -313,21 +338,65 @@ class UploadService {
...
-            file.file.objectKey = await this.putFile(
-                fileUploadURL,
-                file.file.encryptedData
-            );
-
+            if (isDataStream(file.file.encryptedData)) {
+                const { chunkCount, stream } = file.file.encryptedData;
+                const filePartUploadURLs = await this.fetchMultipartUploadURLs(
+                    token,
+                    Math.ceil(chunkCount / 2)
```
Comment (vishnukvmd) [inline src/services/uploadService.ts:383]: Extract `2` to a constant.

File: src/services/uploadService.ts [hunk 4]
```diff
@@ -563,22 +658,85 @@ class UploadService {
...
+        try {
+            const response = await HTTPService.get(
+                `${ENDPOINT}/files/multipart-upload-urls`,
+                {
+                    count,
+                },
+                { 'X-Auth-Token': token }
+            );
+
+            return response.data['multipartUploadUrls'];
```
Comment (vishnukvmd) [inline src/services/uploadService.ts:674]: `URLs`

**Review by vishnukvmd — APPROVED** [review]

File: src/services/uploadService.ts (hunk 2 above)
Comment (Abhinav-grd) [reply src/services/uploadService.ts:153]: Will do this in the sentry integration PR

**Pull request merged**